dev/TeachingTip/TeachingTip.cpp

```diff
@@ -852,6 +902,57 @@ void TeachingTip::OnHeroContentPlacementChanged()
     }
 }
 
+void TeachingTip::OnF6AcceleratorKeyClicked(const winrt::CoreDispatcher&, const winrt::AcceleratorKeyEventArgs& args)
+{
+    if (args.VirtualKey() == winrt::VirtualKey::F6)
+    {
+        auto&& closeButton = m_closeButton.get();
+        auto&& alternateCloseButton = m_alternateCloseButton.get();
+        if (CloseButtonContent())
+        {
+            if (closeButton && closeButton.Visibility() == winrt::Visibility::Visible)
+            {
+                m_closeButtonGettingFocusFromF6Revoker = closeButton.GettingFocus(winrt::auto_revoke, { this, &TeachingTip::OnCloseButtonGettingFocusFromF6 });
+                closeButton.Focus(winrt::FocusState::Keyboard);
+            }
+            else if (alternateCloseButton && alternateCloseButton.Visibility() == winrt::Visibility::Visible)
+            {
+                m_closeButtonGettingFocusFromF6Revoker = alternateCloseButton.GettingFocus(winrt::auto_revoke, { this, &TeachingTip::OnCloseButtonGettingFocusFromF6 });
+                alternateCloseButton.Focus(winrt::FocusState::Keyboard);
+            }
+        }
+        else
+        {
+            if (alternateCloseButton && alternateCloseButton.Visibility() == winrt::Visibility::Visible)
+            {
+                m_closeButtonGettingFocusFromF6Revoker = alternateCloseButton.GettingFocus(winrt::auto_revoke, { this, &TeachingTip::OnCloseButtonGettingFocusFromF6 });
+                alternateCloseButton.Focus(winrt::FocusState::Keyboard);
+            }
+            else if (closeButton && closeButton.Visibility() == winrt::Visibility::Visible)
+            {
+                m_closeButtonGettingFocusFromF6Revoker = closeButton.GettingFocus(winrt::auto_revoke, { this, &TeachingTip::OnCloseButtonGettingFocusFromF6 });
+                closeButton.Focus(winrt::FocusState::Keyboard);
+            }
+        }
+    }
```

Comment:
Can this be rewritten a bit simpler to a function that evaluates which button we care about? Something like this I think would be more readable.

```c++
    auto f6Button = [this]() {
        auto&& firstButton = m_alternateCloseButton.get();
        auto&& secondButton = m_closeButton.get();
        if (CloseButtonContent())
        {
            std::swap(firstButton, secondButton);
        }
        
        if (firstButton && firstButton.Visibility() == winrt::Visibility::Visible)
        {
           return firstButton;
        }

        if (secondButton && secondButton.Visibility() == winrt::Visibility::Visible)
        {
           return secondButton;
        }

        return nullptr;
    }();
    
    if (f6Button)
    {
                m_closeButtonGettingFocusFromF6Revoker = f6Button.GettingFocus(winrt::auto_revoke, { this, &TeachingTip::OnCloseButtonGettingFocusFromF6 });
                f6Button.Focus(winrt::FocusState::Keyboard);
    }
```

Reply:
That is much better, thanks!

---
In reply to: [267992840](https://github.com/Microsoft/microsoft-ui-xaml/pull/469#discussion_r267992840) [](ancestors = 267992840)